dvc/output/local.py

```diff
@@ -61,6 +61,12 @@ def changed(self):
 
         return self.project.cache.local.changed(self.path_info, self.info)
 
+    def changed_cache(self):
+        if not self.use_cache or not self.md5:
```

Comment:
`if not self.use_cache` also applies to all other output types. Also, as you can see, all the `out.changed_cache` methods are only different by the attribute that they use in self.project.cache. It seems like it can be unified nicely in a base class :)

Reply:
@efiop, but the base class is `DependencyBase` I'm not sure if a dependency is "cachable", afair, the only thing DVC caches are outputs :thinking:

Reply:
Yes, we should implement OutputBase and inherit from it as well.

Reply:
I'll look at it at another PR, @efiop , implementing OutputBase would require quite some changes (more suitable for an RFC).